test/server/component.js

```diff
@@ -113,17 +113,20 @@ describe(`Server-side component renderer`, function() {
     const html = el.innerHTML;
     expect(html.toLowerCase()).to.contain(`<div class="attr-app">`);
     expect(html).to.contain(`Value of attribute wombats: 15`);
+    expect(el.attrs.wombats).to.equal(15);
   });
 
   it(`reacts to attribute updates`, async function() {
-    const el = new AttrReflectionApp();
+    const el = new AttrApp();
     el.setAttribute(`wombats`, `15`);
+    expect(el.attrs.wombats).to.equal(15);
```

Comment:
👍 when we go forward with this PR, let's ensure there are some more thorough test cases for `attrsSchema`, for instance testing every type, as well as behavior for invalid values.

Reply:
oh yes. Most definitely. Created a todo list on the PR.

Reply:
This branch has some decent high-level coverage now, but I still feel like I'd like to see at least one chunk of server-side tests that run through the attrs cases a little more systematically and focused rather than as part of bigger rendering tests. For instance with cases divided up by attr type, including both happy path and null/undefined/0/''/malformed-json cases, simply calling `setAttribute()` and checking what ends up in `attrs` without worrying about the rendering cycle.